File: go/remove_duplicates_from_sorted_array.go
package algos

func removeDuplicates(nums []int) int {
	i := 0
	for j := 0; j < len(nums); j++ {
		if nums[i] != nums[j] {
			i += 1
			nums[i] = nums[j]
		}
	}
	return i + 1
}

File: rust/src/algos/remove_duplicates_from_sorted_array.rs
pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
    let mut slow: usize = 0;
    let mut fast: usize = 1;

    while fast < nums.len() {
        while fast < nums.len() && nums[slow] == nums[fast] {
            fast += 1;
        }
        if fast < nums.len() {
            slow += 1;
            nums[slow] = nums[fast];
        }
    }
    (slow + 1) as i32
}
